.github/workflows/docs.yaml

```diff
@@ -0,0 +1,76 @@
+on:
+  push:
+    branches:
+      - docs-workflow
+    tags-ignore:
+      - "**-rc**"
+
+name: Deploy Datafusion Python site
+
+jobs:
+  build-docs:
+    name: Build docs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set target branch
+        id: target-branch
+        run: |
+          set -x
+          if test '${{ github.ref }}' = 'refs/heads/docs-workflow'; then
+            echo "value=asf-staging" >> $GITHUB_OUTPUT
+          elif test '${{ github.ref_type }}' = 'tag'; then
+            echo "value=asf-site" >> $GITHUB_OUTPUT
+          else
+            echo "Unsupported input: ${{ github.ref }} / ${{ github.ref_type }}"
+            exit 1
+          fi
+      - name: Checkout docs sources
+        uses: actions/checkout@v3
+      - name: Checkout docs target branch
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ steps.target-branch.outputs.value }}
+          path: docs-target
```

Comment:
It seems that we need to prepare `asf-site` and `asf-staging` branches for this: https://github.com/martin-g/arrow-datafusion-python/actions/runs/3633415170/jobs/6130403408

How about using `git worktree` instead?

```diff
diff --git a/.github/workflows/docs.yaml b/.github/workflows/docs.yaml
index 806202a..bfafafa 100644
--- a/.github/workflows/docs.yaml
+++ b/.github/workflows/docs.yaml
@@ -26,12 +26,6 @@ jobs:
           fi
       - name: Checkout docs sources
         uses: actions/checkout@v3
-      - name: Checkout docs target branch
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-          ref: ${{ steps.target-branch.outputs.value }}
-          path: docs-target
       - name: Setup Python
         uses: actions/setup-python@v4
         with:
@@ -61,7 +55,12 @@ jobs:
       - name: Copy & push the generated HTML
         run: |
           set -x
+          git worktree add docs-target
           cd docs-target
+          if ! git checkout ${{ steps.target-branch.outputs.value }}; then
+            git checkout --orphan ${{ steps.target-branch.outputs.value }}
+            git rm -rf .
+          fi
           # delete anything but: 1) '.'; 2) '..'; 3) .git/
           find ./ | grep -vE "^./$|^../$|^./.git" | xargs rm -rf
           cp ../.asf.yaml .
```

Reply:
IMO this is more complex/confusing than the current solution.

Reply:
A task before merging this PR is to create `asf-staging` branch.

Reply:
Created: https://github.com/apache/arrow-datafusion-python/tree/asf-staging